dev_docs/deep_zoom.md

```diff
@@ -0,0 +1,31 @@
+# Spatial Mode Deep Zoom Feature
+
+The Deep Zoom feature in CELLxGENE Explorer enhances the spatial exploration of datasets by allowing users to zoom in and out of high-resolution images smoothly. This functionality is critical for detailed examination of spatial data, such as those from Slide-seq V2 or Visium, providing fine-grained insight into tissue samples and cellular interactions.
+
+It is built on the OpenSeadragon (OSD) JavaScript library, integrated into the Explorer’s frontend to deliver a responsive zooming experience. The backend handles the generation and storage of deep zoom image tiles, making it efficient to load and view high-resolution spatial data across environments (dev, staging, prod).
+
+## Frontend Integration (Explorer FE)
+
+- [OpenSeadragon (OSD)](https://openseadragon.github.io/): The OSD library is imported into the frontend to handle interactive zoom and pan operations. OSD dynamically loads image tiles based on the user’s zoom level and position.
+- Zoom & Pan Synchronization: Zooming and panning are kept in sync with spatial layers (e.g., dot layers) in the Explorer to ensure smooth interaction across different dataset views.
+- Dynamic Resource Loading: URLs for the image tiles are generated based on the current environment (local, dev, staging, prod), ensuring the correct assets are fetched dynamically.
```

Comment:
For local development we still use the tiles from dev/staging S3! E.g., 

https://cellxgene.dev.single-cell.czi.technology/spatial-deep-zoom/super-cool-spatial/spatial_files/10/1_3.jpeg

https://cellxgene.staging.single-cell.czi.technology/spatial-deep-zoom/super-cool-spatial/spatial_files/10/1_3.jpeg

```suggestion
- Dynamic Resource Loading: URLs for the image tiles are generated based on the current environment (dev, staging, prod), ensuring the correct assets are fetched dynamically.
```

<img width="1337" alt="Screenshot 2024-09-11 at 12 38 25 PM" src="https://github.com/user-attachments/assets/fc05f6f5-3a55-4cf7-9bbb-c606d7393826">